appcelerator/titanium_mobile_windows
# [TIMOB-24093] Hyperloop for Windows (Reflection based)

Ready for review.

[TIMOB-24093](https://jira.appcelerator.org/browse/TIMOB-24093)

Hyperloop for Windows. Basically removes previous Hyperloop (static version) and introduces reflection-based Hyperloop.

This needs Appcelerator Platform Pro account as well as `hyperloop` module for Windows.

# How to Test

- Download Titanium SDK build artifacts from here: [titanium_mobile_windows_10_prs/1249/artifact/dist/windows](http://studio-jenkins.appcelerator.org/job/titanium_mobile_windows_10_prs/1249/artifact/dist/windows/) as well as [titanium_mobile_windows_8.1_prs/991/artifact/dist/windows](http://studio-jenkins.appcelerator.org/job/titanium_mobile_windows_8.1_prs/991/artifact/dist/windows/), and extract/replace them with your Titanium SDK `windows` directory.
- Make sure you have installed `hyperloop` module and you have Appcelerator Platform Pro account

```javascript
var win = Ti.UI.createWindow({ backgroundColor: 'green' });
win.addEventListener('open', function () {

    // Test instance method call
    var System_Object = require('System.Object');
    var object = new System_Object();
    alert(object.GetHashCode());
});
win.open();
```

In addition to that you'll need to set hyperloop module in your `tiapp.xml`.

```xml
<modules>
  <module platform="windows">hyperloop</module>
</modules>
```
